### PR TITLE
feat: [CI-3255]: Add secret volumes support on delegate to support self signed certs

### PR DESCRIPTION
**Pull request by vistaarjuneja**
…ned certs

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30788)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java [hunk 1]
```diff
@@ -207,6 +228,74 @@ private void createImageSecrets(CoreV1Api coreV1Api, String namespace, CIK8PodPa
     log.info("Image secret creation took: {} for pod: {} ", timer.stop(), podParams.getName());
   }
 
+  private void updatePodWithDelegateVolumes(CoreV1Api coreV1Api, String namespace, V1Pod pod) {
+    List<V1Volume> podVolumes = new ArrayList<>();
+    List<V1VolumeMount> containerVolumeMounts = new ArrayList<>();
+
+    Map<String, List<String>> srcDestMappings = secretVolumesHelper.getSrcDestMappings();
```
Comment (harshjain12) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:235]: can we rename getSrcDestMappings
I am not able to understand this method just by reading name

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java [hunk 2]
```diff
@@ -207,6 +228,74 @@ private void createImageSecrets(CoreV1Api coreV1Api, String namespace, CIK8PodPa
...
+
+      // Create secrets, pod volumes, and container volume mounts for all containers in the spec
+      V1Secret secret =
+          new V1SecretBuilder()
+              .withMetadata(new V1ObjectMetaBuilder().withNamespace(namespace).withName(secretKey).build())
+              .withData(ImmutableMap.of(secretKey, content.getBytes()))
+              .withType(OPAQUE_SECRET_TYPE)
+              .build();
+
+      cik8JavaClientHandler.createOrReplaceSecret(coreV1Api, secret, namespace);
```
Comment (harshjain12) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:264]: Wr are creating secret one by one.
Can we use batching ?
Initialize step is already slow

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/helper/SecretVolumesHelper.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class SecretVolumesHelper {
+  public static final String CI_MOUNT_VOLUMES = "CI_MOUNT_VOLUMES";
+
+  // Reads the list of volumes to be mounted on the build containers and returns
+  // a mapping from source paths to a list of destination paths.
+  public Map<String, List<String>> getSrcDestMappings() {
```
Comment (harshjain12) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/helper/SecretVolumesHelper.java:27]: refactor this method and divide into small methods

**Comment by harshjain12** [comment]
UTs are missing

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java (hunk 1 above)
Comment (vistaarjuneja) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:235]: Done, added more comments as well

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java (hunk 2 above)
Comment (vistaarjuneja) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:264]: If by batching, you mean sending one call to create all the secrets - I could not find support for such an operation in this library.
If you mean creating a single secret with all these values present as key value pairs, there is a limitation with that which will not let us mount the same source path to two different destination paths.

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/helper/SecretVolumesHelper.java (hunk 3 above)
Comment (vistaarjuneja) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/helper/SecretVolumesHelper.java:27]: Done

**Review by harshjain12 — APPROVED** [review]
> Reviewed 1 of 5 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30788)* status: :shipit: complete! ✓

**Pull request merged**